Dockerfile

```diff
@@ -0,0 +1,31 @@
+FROM docker.ocf.berkeley.edu/theocf/debian:stretch
+
+RUN apt-get update \
+    && DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends \
+        libpam-krb5 \
+        nginx \
+        python3 \
+        python3-pip \
+        python3-setuptools \
+        runit \
+    && apt-get clean \
+    && rm -rf /var/lib/apt/lists/*
+
+COPY services/nginx/nginx.conf /etc/nginx
+COPY discourse_pam /etc/pam.d/discourse
+
+RUN mkdir -p /opt/discourse_sso
+
+COPY requirements.txt /opt/discourse_sso/requirements.txt
+RUN pip3 install -r /opt/discourse_sso/requirements.txt
+
+RUN chown nobody:nogroup /opt/discourse_sso
+WORKDIR /opt/discourse_sso
+USER nobody
+
+RUN echo "ocfstaff\nopstaff" > /opt/discourse_sso/allowed-groups
+
+COPY --chown=nobody:nogroup app.py /opt/discourse_sso
+COPY --chown=nobody:nogroup services /opt/discourse_sso/services
```

Comment:
I would reorder these things to try and take advantage of Docker's layer caching as much as possible, possibly like so, assuming the requirements are the things that are going to be updated most frequently.

```Dockerfile
COPY services/nginx/nginx.conf /etc/nginx
COPY discourse_pam /etc/pam.d/discourse

RUN install -g nogroup -o nobody -d /opt/discourse_sso
RUN echo "ocfstaff\nopstaff" > /opt/discourse_sso/allowed-groups

COPY --chown=nobody:nogroup app.py /opt/discourse_sso
COPY --chown=nobody:nogroup services /opt/discourse_sso/services

WORKDIR /opt/discourse_sso
USER nobody

COPY requirements.txt /opt/discourse_sso/requirements.txt
RUN pip3 install -r /opt/discourse_sso/requirements.txt

CMD ["runsvdir", "/opt/discourse_sso/services"]
```

Reply:
This won't quite work, since we need to do the `pip3 install` as the root user, not `nobody`. So it has to happen before changing the `USER`.

However, I'll look into ways of re-ordering this to take advantage of layer caching.

Reply:
I reordered this but let me know if you think there's a better way.